### PR TITLE
Add built-in language grammar extensions for C#, F#, HTML, and Javascript.

### PR DESCRIPTION
**Pull request by corivera**
.NET Interactive adds C#, F#, HTML, and Javascript support to notebooks, so we need to bring in these VS Code language extensions to get the correct language grammars for those new languages (i.e. text coloring in the editor).

I also had to make a small fix in the git hygiene script to correctly handle filenames with spaces, since one of the javascript grammar files has spaces in its name.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
@kburtram What are your thoughts on having these language extensions included in core ADS? While it doesn't really make sense to have them outside of .NET Interactive they are at least pretty small.

(the alternative would be to separate them out into standalone extensions that are then installed through the gallery)

File: build/hygiene.js [hunk 1]
```diff
@@ -195,7 +195,7 @@ function createGitIndexVinyls(paths) {
 				}
 
 				cp.exec(
-					`git show :${relativePath}`,
+					`git show :"${relativePath}"`,
```
Comment (Charles-Gagnon) [inline build/hygiene.js:198]: Looks like VS Code has a slightly different fix, would suggest using this one instead so we keep in sync with their stuff

https://github.com/microsoft/vscode/blob/main/build/hygiene.js#L234

Comment (corivera) [reply build/hygiene.js:198]: That line still throws the same error, since the windows path isn't quoted.

Comment (Charles-Gagnon) [reply build/hygiene.js:198]: I'm curious why they don't have this same issue then...

At the very least let's copy over what they have and add the extra quotes around the windows one.

Comment (corivera) [reply build/hygiene.js]: That javascript regex file is the only spaced file I've seen, so maybe they don't change it that often.

**Review by Charles-Gagnon — APPROVED** [review]
Looks good to me pending @kburtram's approval

File: build/hygiene.js [hunk 2]
```diff
@@ -195,7 +195,7 @@ function createGitIndexVinyls(paths) {
 				}
 
 				cp.exec(
-					`git show :${relativePath}`,
+					process.platform === 'win32' ? `git show :"${relativePath}"` : `git show ':${relativePath}'`, // {{SQL CARBON EDIT}} Add double quotes around Windows path
```
Comment (kburtram) [inline build/hygiene.js:198]: any idea why they don't need this upstream in vscode repo?

**Review by kburtram — APPROVED** [review]
LGTM though seems strange to need to update the shared build scripts when copying code from vscode repo.  i.e. if the same code exists in vscode repo why doesn't the hygiene task have the same problem there?

File: build/hygiene.js (hunk 2 above)
Comment (corivera) [reply build/hygiene.js:198]: No idea, but I couldn't check in that javascript regex file without it. Going by the history, this whole part used to be single quoted, but then was changed to have no quotes on windows to fix some other kind of issue. The javascript regex file hasn't changed in 6 years, so maybe they haven't hit issues with any other filenames with spaces. That's certainly the only one I've seen while working on ADS the past few years.

Comment (corivera) [reply build/hygiene.js:198]: If you want, I could just revert this file. The only thing that was being blocked was comitting that one javascript file.

Comment (Charles-Gagnon) [reply build/hygiene.js:198]: Yeah let's do that - if it comes up again we can look into it further but it seems extremely unlikely to happen in most cases

**Pull request merged**